pkg/manifest/manifest.go

```diff
@@ -318,13 +320,58 @@ func (manifest *manifest) Marshal() ([]byte, error) {
 }
 
 func (manifest *manifest) WriteConfig(out io.Writer) error {
-	bytes, err := yaml.Marshal(manifest.Config)
+	return writeConfig(manifest.ID(), manifest.Config, out)
+}
+
+// MergedConfig merges this manifest's config with the given map, returning a new config map.
+// The config of the manifest is not modified.
+func (manifest *manifest) MergedConfig(with map[interface{}]interface{}) (map[interface{}]interface{}, error) {
+	merged := manifest.GetConfig()
+
+	// Tried just marshalling the second config, then unmarshalling it into the same map.
+	// However, that doesn't preserve inner maps correctly.
+	mergeInto(merged, with)
+
+	return merged, nil
```

Comment:
says it can error, but in no code path will there be a non-nil error.

I used to say it is an error to merge a non-map with a map, but then I just ended up overwriting everything and not showing errors. Wonder if anyone can think of any time to error. If not, maybe I just won't return an error.